herddb-collections/src/main/java/herddb/collections/CollectionsManager.java

```diff
@@ -0,0 +1,320 @@
+                new MemoryCommitLogManager(), tmpDirectory,
+                new ServerHostData("localhost", 0, "", false, Collections.emptyMap()),
+                configuration, NullStatsLogger.INSTANCE);
+        server.setMaxDataUsedMemory(maxMemory);
+    }
+
+    public void start() throws Exception {
+        server.start();
+        server.waitForTablespace(TableSpace.DEFAULT, 60000);
+        tableSpaceManager = server.getTableSpaceManager(TableSpace.DEFAULT);
+    }
+
+    @Override
+    public void close() {
+        server.close();
+    }
+
+    public <V> TmpMapBuilder<V> newMap() {
+        return new TmpMapBuilder<>();
+    }
+
+    private void createTable(String tmpTableName, int pkType) throws StatementExecutionException {
+        Table table = Table
+                .builder()
+                .name(tmpTableName)
+                .column("pk", pkType)
+                // no need to define other columns
+                .primaryKey("pk")
+                .tablespace(TableSpace.DEFAULT)
+                .build();
+        CreateTableStatement createTable = new CreateTableStatement(table);
+        tableSpaceManager.executeStatement(createTable, new StatementEvaluationContext(),
+                TransactionContext.NO_TRANSACTION);
+    }
+
+    public static final class Builder {
+
+        private long maxMemory = ServerConfiguration.PROPERTY_MEMORY_LIMIT_REFERENCE_DEFAULT;
+        private Path tmpDirectory;
+
+        /**
+         * Max memory to use. This is an upper bound to the amount of memory directly referenced by the
+         * CollectionsManager. The system will automatically swap to disk data in order to respect this limit.
+         *
+         * @param maxMemory the amount of memory, in bytes. If not set it will use HerdDB defaults
+         * @return the build itself
+         */
+        public Builder maxMemory(long maxMemory) {
+            this.maxMemory = maxMemory;
+            return this;
+        }
+
+        /**
+         * The directory to store the database. It is expected that the directory is empty.
+         *
+         * @param tmpDirectory
+         * @return the build itself
+         */
+        public Builder tmpDirectory(Path tmpDirectory) {
+            this.tmpDirectory = tmpDirectory;
+            return this;
+        }
+
+        /**
+         * Creates the CollectionsManager. You must call {@link CollectionsManager#start() }
+         * in order to boot the system.
+         *
+         * @return the new not-yet-started CollectionsManager.
+         */
+        public CollectionsManager build() {
+            return new CollectionsManager(maxMemory, tmpDirectory);
+        }
+
+    }
+
+    public class TmpMapBuilder<V> {
+
+        private ValueSerializer<V> valueSerializer = DEFAULT_VALUE_SERIALIZER;
+        private int expectedValueSize = 64;
+
+        public class IntTmpMapBuilder<VI extends V> {
```

Comment:
Do we need to add LongTmpMapBuilder? We support NOTNULL_LONG key type.

Reply:
This is the first stub for this new sub project.
I guess we will add support for 'long' soon.
I can do it now if you think it is useful.
In the cases we will cover in MagNews we only have ints or Pojos.
Supporting longs will be very easy.
I will do.
I feel we won't ever support timestamps, we are not going to create a time series DB